pkg/config/reconciler.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package config
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+// Reconciler is responsible for reconciling Apps.
+type Reconciler struct {
+	coreClient kubernetes.Interface
+	log        logr.Logger
+}
+
+// NewReconciler constructs new Reconciler.
+func NewReconciler(coreClient kubernetes.Interface, log logr.Logger) *Reconciler {
+	return &Reconciler{coreClient, log}
+}
+
+var _ reconcile.Reconciler = &Reconciler{}
+
+// AttachWatches configures watches needed for reconciler to reconcile Apps.
+func (r *Reconciler) AttachWatches(controller controller.Controller) error {
+	// only reconcile on the KC's config
+	p := predicate.NewPredicateFuncs(func(o client.Object) bool {
+		return o.GetName() == kcConfigName
+	})
+
+	err := controller.Watch(&source.Kind{Type: &v1.ConfigMap{}}, &handler.EnqueueRequestForObject{}, p)
+	if err != nil {
+		return fmt.Errorf("Watching Configmaps: %s", err)
+	}
+
+	err = controller.Watch(&source.Kind{Type: &v1.Secret{}}, &handler.EnqueueRequestForObject{}, p)
+	if err != nil {
+		return fmt.Errorf("Watching Secrets: %s", err)
+	}
```

Comment:
are these somehow limited to kapp controller namespace?

Reply:
no, they are currently not scoped by namespace at all. If the config must be in the `kapp-controller` namespace I can update this so that it filters by NS 👍

Reply:
I changed this so that it follows the same logic as `GetConfig()`, i.e it loads the namespace from the kubeconfig.